src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -12,37 +14,194 @@ internal static partial class ClrThreadPool
         
         private static int s_minThreads; // TODO: Initialize
         private static int s_maxThreads; // TODO: Initialize
+        private static readonly LowLevelLock s_maxMinThreadLock = new LowLevelLock();
+        
+        private static readonly LowLevelLock s_threadAdjustmentLock = new LowLevelLock();
+        private static ThreadCounts s_counts = new ThreadCounts();
+        private static int s_lastDequeueTime;
+        private static int s_priorCompletedWorkRequestsTime;
+        private static int s_nextCompletedWorkRequestsTime;
+        private static long s_currentSampleStartTime;
+        private static int s_priorCompletionCount = 0;
```

Comment:
Please also look at the DECLSPEC_ALIGN used for some of these fields to ensure that they are in their own cache line